djangocms_link/models.py

```diff
@@ -1,98 +1,153 @@
+    )
+    # advanced options
+    target = models.CharField(
+        verbose_name=_('Target'),
+        choices=TARGET_CHOICES,
+        blank=True,
+        max_length=255,
     )
-    anchor = models.CharField(_('anchor'), max_length=128, blank=True,
-                              help_text=_('This applies only to page and text links.'
-                                          ' Do <em>not</em> include a preceding "#" symbol.'))
-    # Explicitly set a max_length so that we don't end up with different
-    # schemata on Django 1.7 vs. 1.8.
-    mailto = models.EmailField(_('email address'), max_length=254, blank=True, null=True,
-                               help_text=_('An email address has priority over a text link.'))
-    phone = models.CharField(_('Phone'), blank=True, null=True, max_length=40,
-                             help_text=_('A phone number has priority over a mailto link.'))
-    target = models.CharField(_('target'), blank=True, max_length=100,
-                              choices=TARGET_CHOICES)
-
-    # These attributes are already managed by other fields
-    EXCLUDED_KEYS = [
-        'href',
-        'target',
-    ]
-
     attributes = AttributesField(
-        _('link tag attributes'), excluded_keys=EXCLUDED_KEYS, blank=True,
-        help_text=_('Optional. Link HTML tag attributes'),
+        verbose_name=_('Attributes'),
+        blank=True,
+        excluded_keys=['href', 'target'],
+    )
+
+    # Add an app namespace to related_name to avoid field name clashes
+    # with any other plugins that have a field with the same name as the
+    # lowercase of the class name of this model.
+    # https://github.com/divio/django-cms/issues/5030
+    cmsplugin_ptr = models.OneToOneField(
+        CMSPlugin,
+        related_name='%(app_label)s_%(class)s',
+        parent_link=True,
     )
 
     class Meta:
         abstract = True
 
     def __str__(self):
-        return self.name
+        return self.name or str(self.pk)
+
+    def get_short_description(self):
+        if self.name:
+            return '{} ({})'.format(self.name, self.get_link())
+        return self.get_link()
 
-    def link(self):
+    def get_link(self):
         if self.phone:
-            link = 'tel:%s' % self.phone
+            link = 'tel:{}'.format(self.phone.replace(' ', ''))
         elif self.mailto:
-            link = 'mailto:%s' % self.mailto
-        elif self.url:
-            link = self.url
-        elif self.page_link_id:
-            link = self.page_link.get_absolute_url()
+            link = 'mailto:{}'.format(self.mailto)
+        elif self.external_link:
+            link = self.external_link
+        elif self.internal_link_id:
+            link = self.internal_link.get_absolute_url()
         else:
             link = ''
-        if (self.url or self.page_link or not link) and self.anchor:
-            link += '#%s' % self.anchor
+        if (self.external_link or self.internal_link_id or not link) and self.anchor:
```

Comment:
I think the condition is more readable as `(not self.phone and not self.mailto) and self.anchor`